### PR TITLE
Add permanent redirect from HTTP to HTTPS

### PR DESCRIPTION
**Pull request by irahopkinson**
Configure permanent redirect from HTTP to HTTPS.

### CODE REVIEW

File: templates/vhost_proxypath_ssl.conf.j2 [hunk 1]
```diff
@@ -1,46 +1,53 @@
 {% for host in item.virtual_hosts %}
 <VirtualHost *:{{ host.port | default('80') }}>
     ServerName {{ item.server_name }}
-{% if host.server_alias is defined %}    ServerAlias {{ host.server_alias | join(' ') }}
+{% if host.server_alias is defined %}
+    ServerAlias {{ host.server_alias | join(' ') }}
 {% endif %}
```
Comment (cambell-prince) [inline templates/vhost_proxypath_ssl.conf.j2]: The layout here, and throughout the file, is intentional.  This isn't code, the white space is important outside of the {% %} scope.  Your change would force a blank line between the ServerName and ServerAlias directives.  No harm, but not typically done in most Apache config files.

I suggest keeping the formatting as it was for this reason.

File: templates/vhost_proxypath_ssl.conf.j2 [hunk 2]
```diff
@@ -1,46 +1,53 @@
...
-{% if item.server_admin is defined %}    ServerAdmin {{ item.server_admin }}
+{% if item.server_admin is defined %}
+    ServerAdmin {{ item.server_admin }}
```
Comment (cambell-prince) [inline templates/vhost_proxypath_ssl.conf.j2]: Formatting as above.

File: templates/vhost_proxypath_ssl.conf.j2 [hunk 3]
```diff
@@ -1,46 +1,53 @@
...
 {% endif %}
 
 {% if host.has_ssl %}
     SSLEngine On
     SSLCertificateFile {{host.ssl_path | default(apache_ssl_path)}}/{{host.certificate_file}}
     SSLCertificateKeyFile {{host.ssl_path | default(apache_ssl_path)}}/{{host.key_file}}
-{% if host.chain_file is defined %}    SSLCertificateChainFile {{host.ssl_path | default(apache_ssl_path)}}/{{ host.chain_file }}
+{% if host.chain_file is defined %}
+    SSLCertificateChainFile {{host.ssl_path | default(apache_ssl_path)}}/{{ host.chain_file }}
```
Comment (cambell-prince) [inline templates/vhost_proxypath_ssl.conf.j2]: Formatting as above

File: templates/vhost_proxypath_ssl.conf.j2 [hunk 4]
```diff
@@ -1,46 +1,53 @@
...
 {% endif %}
     
 {% endif %}
+{% if not host.has_ssl and host.has_redirect_permanent is defined and host.has_redirect_permanent %}
+    Redirect permanent / https://{{ item.server_name }}/
+{% else %}
     DocumentRoot {{ item.document_root }}
-{% if host.extra is defined %}
+    {% if host.extra is defined %}
```
Comment (cambell-prince) [inline templates/vhost_proxypath_ssl.conf.j2]: I think its better to have the template directives start in column 0.  Its not code, and as above white space outside of {% %} matters.

**Review by cambell-prince — CHANGES_REQUESTED** [review]
Thanks @irahopkinson Main comment is about the meaningful white space (which is annoying for sure).  Otherwise fine.  Thanks for the PR.

**Comment by irahopkinson** [comment]
Ha, for the first three VSCode was being more helpful with the file layout than I wanted it to be. I've also removed the indents.

**Pull request merged**